diff --git a/qml/BezierCanvas.qml b/qml/BezierCanvas.qml
index 5e68cfa..edf0bf7 100644
--- a/qml/BezierCanvas.qml
+++ b/qml/BezierCanvas.qml
@@ -55,7 +55,7 @@ Canvas {
         var secondsUnit = secondTotal % 10;
         var secondsTen = (secondTotal % 100 - secondTotal % 10) / 10;
         var secondsUnitRatio = millis / 1000.0;
-        var secondsTenRatio = (secondsUnit * 1000 + millis) / 10000.0;
+        var secondsTenRatio = (secondsUnit * 1000 + millis) / 10000.0;/*
         setup.init.secondsUnitsAnimator.animationStartRatio = CoreFunctions.getAnimStartRatio(1.0, setup.animDurationUser);
         setup.init.secondsUnitsAnimator.update(context,
                                           CanvasFunctions.determineDigit(secondsUnit),
@@ -65,7 +65,7 @@ Canvas {
         setup.init.secondsTensAnimator.update(context,
                                          CanvasFunctions.determineDigit(secondsTen),
                                          CanvasFunctions.determineDigit(CoreFunctions.getNextInt(secondsTen, 5)),
-                                         secondsTenRatio);
+                                         secondsTenRatio);*/
 
         // Blink Dots
         if (setup.showDots && setup.blinkDots) {
diff --git a/qml/BezierInit.qml b/qml/BezierInit.qml
index 808e909..dca30cf 100644
--- a/qml/BezierInit.qml
+++ b/qml/BezierInit.qml
@@ -40,14 +40,14 @@ Item {
 
     BezierAnimator { // X0:00:00
         id: _hoursTensAnimator
-        origX: 0.0 * visualScaling
+        origX: 300.0 * visualScaling
         origY: yOff * visualScaling
         animationStartRatio: 35995.0 / (35995.0 + 5.0)
     }
 
     BezierAnimator { // 0X:00:00
         id: _hoursUnitsAnimator
-        origX: 300.0 * visualScaling
+        origX: 600.0 * visualScaling
         origY: yOff * visualScaling
         renderDots: true
         animationStartRatio: 3595.0 / (3595.0 + 5.0)
@@ -55,16 +55,16 @@ Item {
 
     BezierAnimator { // 00:X0:00
         id: _minutesTensAnimator
-        origX: 800.0 * visualScaling
+        origX: 1100.0 * visualScaling
         origY: yOff * visualScaling
         animationStartRatio: 595.0 / (595.0 + 5.0)
     }
 
     BezierAnimator { // 00:0X:00
         id: _minutesUnitsAnimator
-        origX: 1100.0 * visualScaling
+        origX: 1400.0 * visualScaling
         origY: yOff * visualScaling
-        renderDots: true
+        // renderDots: true
         animationStartRatio: 55.0 / (55.0 + 5.0)
     }
 
