From 6673c7f60ec72b5f78ebda20f680c49375a658c9 Mon Sep 17 00:00:00 2001
From: EXL <exlmotodev@gmail.com>
Date: Sat, 27 Oct 2018 02:51:04 +0700
Subject: [PATCH] Haiku OS: Fix opening links in the package version

---
 engine/common/system.c | 23 ++++++++++++++---------
 1 file changed, 14 insertions(+), 9 deletions(-)

diff --git a/engine/common/system.c b/engine/common/system.c
index 8929dc0b..e455002c 100644
--- a/engine/common/system.c
+++ b/engine/common/system.c
@@ -376,19 +376,11 @@ void Sys_ShellExecute( const char *path, const char *parms, qboolean shouldExit
 					document.location.href = Pointer_stringify($0);
 				return 0;
 			}, (int)path );
-#elif (defined(__linux__) && !defined (__ANDROID__)) || defined (__FreeBSD__) || defined (__NetBSD__) || defined(__OpenBSD__) || defined(__APPLE__) || defined(__HAIKU__)
+#elif (defined(__linux__) && !defined (__ANDROID__)) || defined (__FreeBSD__) || defined (__NetBSD__) || defined(__OpenBSD__) || defined(__APPLE__)
 
 	if( !Q_strcmp( path, GENERIC_UPDATE_PAGE ) || !Q_strcmp( path, PLATFORM_UPDATE_PAGE ))
 		path = XASH_UPDATE_PAGE;
 
-	// Prevent "open: www.url.com: No such file or directory" error
-	char http[MAX_SYSPATH];
-	if( Q_strstr( path, "www" ) && Q_strncmp( path, "http", 4 ) )
-	{
-		Q_snprintf( http, MAX_SYSPATH, "%s%s", "http://", path );
-		path = http;
-	}
-
 	char xdgOpen[128];
 	if( Sys_FindExecutable( OPEN_COMMAND, xdgOpen, sizeof( xdgOpen ) ) )
 	{
@@ -404,6 +396,19 @@ void Sys_ShellExecute( const char *path, const char *parms, qboolean shouldExit
 	else MsgDev( D_WARN, "Could not find "OPEN_COMMAND" utility\n" );
 #elif defined(__ANDROID__) && !defined(XASH_DEDICATED)
 	Android_ShellExecute( path, parms );
+#elif defined(__HAIKU__)
+	// Prevent "open: www.url.com: No such file or directory" error
+	char http[MAX_SYSPATH];
+	if( Q_strncmp( path, "http", 4 ) )
+	{
+		Q_snprintf( http, MAX_SYSPATH, "%s%s", "http://", path );
+		path = http;
+	}
+
+	// This will work in both package and standalone versions
+	char command[MAX_SYSPATH];
+	Q_snprintf( command, MAX_SYSPATH, "%s %s &", OPEN_COMMAND, path );
+	system( command );
 #endif
 
 	if( shouldExit )
-- 
2.19.1

