diff --git a/cr3qt/CMakeLists.txt b/cr3qt/CMakeLists.txt
index 80395df..4fd6ef2 100644
--- a/cr3qt/CMakeLists.txt
+++ b/cr3qt/CMakeLists.txt
@@ -148,8 +148,6 @@ if(CMAKE_GENERATOR MATCHES "Visual Studio.*")
 endif(CMAKE_GENERATOR MATCHES "Visual Studio.*")
 endif(WIN32)
 
-
-
 IF(MAC)
     message("Qt libs = ${QT_LIBRARIES}")
 #    SET(FRAMEWORKS "-framework ApplicationServices -framework Carbon -framework AppKit")
@@ -157,7 +155,17 @@ IF(MAC)
     SET (EXTRA_LIBS ${QT_LIBRARIES} ${STD_LIBS} )
 #${QT_LIBRARIES} 
 ELSEIF (UNIX)
-    SET (EXTRA_LIBS ${QT_LIBRARIES} fontconfig ${STD_LIBS} )
+    IF(DEFINED CMAKE_TOOLCHAIN_FILE)
+      message("Warning: Cross-compilation detected!")
+      IF(NOT DEFINED FONTCONFIG_LIB)
+	message("Error: Please define FONTCONFIG_LIB variable, eg: -DFONTCONFIG_LIB=/usr/lib/libfontconfig.so!")
+      ENDIF(NOT DEFINED FONTCONFIG_LIB)
+      message("Warning: Fontconfig library here: " ${FONTCONFIG_LIB})
+      SET (EXTRA_LIBS ${QT_LIBRARIES} ${FONTCONFIG_LIB} ${STD_LIBS} )
+    ELSE()
+      message("Warning: libfontconfig will be linked!")
+      SET (EXTRA_LIBS ${QT_LIBRARIES} fontconfig ${STD_LIBS} )
+    ENDIF(DEFINED CMAKE_TOOLCHAIN_FILE)
 ELSE()
     SET (EXTRA_LIBS ${STD_LIBS} ${QT_LIBRARIES})
 ENDIF(MAC)
